### PR TITLE
Feature/create lazy element robert w

### PR DESCRIPTION
**Pull request by row49382**
**jmaqs-selenium**

**Lazy Element**
* First implementation of Lazy Element parity
* Wrapped all element find methods in the getElement method so that we are respecting the cache when doing so
* Updated all suppliers for finding elements that use the parent element, to use FluentWait instead of UIWait. This is because UIWait only takes in the WebDriver so we lose the parent context. FluentWait allows us to pass in the WebElement which retains the parent information.
* Updating tests to use getText instead of getValue for the Footer lazy element. This is because the footer doesn't contain a value attribute so it would always timeout
* Updated tests that use finds to assert using not same instead of not equals. This is because the RemoteWebElement overrides equals to compare ids, which is cached based on the selector used to find the element.
* Added overload to FluentWaitFactory that uses the default waiting values from the config.
* Adjusting code smells from Sonar Lint
* Added unit tests for the Lazy Element

**Lazy Element Enhancements Requested**
* Renaming AbstractLazyWebElement to AbstractLazyElement since we plan to use this as a base for MobileLazyElement.
* Created methods in LazyWebElement findElement and findElements that get the WebElements and maps them to LazyWebElement
* Reorganized constructor params to be the same as MAQS implementation
* Updated javadocs to be consistent
* Created method to get the WebElement with the newly created element index
* resolving issue with passing null for Integer and converting to int
* adding tests to validate new functions

### CODE REVIEW

**Comment by billros** [comment]
@Jason Edstrom<mailto:JasonE@magenic.com>, what is our “parity plan”?

There is a flaw on the 5.x version of FluentElement that we are fixing in 6.
Do we want to implement it now, rather than later?

Sspecifically FindElements which returns webelements, not FluentElements.  This means we can never get a list of child elements as FluentElements…

In 6.0 we are making a (breaking) change.  FindElement and FindElements methods on a FluentElement return FluentElements.
FindRawElement(s) return IWebElements.

See https://github.com/Magenic/MAQS/issues/312

**Comment by jason-edstrom** [comment]
@billros I think that is a great idea.  I would rather make the changes now and release it in full.  @row49382, do you have bandwidth to add those changes now that the pr build is passing?

File: jmaqs-selenium/src/main/java/com/magenic/jmaqs/selenium/factories/FluentWaitFactory.java [hunk 1]
```diff
@@ -1,16 +1,18 @@
 /*
- * Copyright 2020 (C) Magenic, All rights Reserved
+ * Copyright 2019 (C) Magenic, All rights Reserved
```
Comment (jonreding2010) [inline jmaqs-selenium/src/main/java/com/magenic/jmaqs/selenium/factories/FluentWaitFactory.java:2]: update copyright to 2020

**Comment by row49382** [comment]
> @billros I think that is a great idea. I would rather make the changes now and release it in full. @row49382, do you have bandwidth to add those changes now that the pr build is passing?
I can make these updates today now that I'm on bench. 

@billros @jason-edstrom 
So that this is clear what the request is:
Add two separate methods:
- `findRawElement()` and `findRawElements()`
    - The implementation for these will be what we have currently for `findElement()` and `findElements()` respectively
- refactor the original `findElement()` and `findElements()` method to return the `WebElement` interfaced elements you would normally see from those methods to be `LazyElement` instead
- Re-name the following methods to have the prefix `getTheRaw...` (or in the last case: `getRaw...`)
   - `getTheVisibleElement()`
   - `getTheClickableElement()`
   - `getTheExistingElement()`
   - `getIndexed()`
- Update/write unit tests for the new methods

Does that look correct?

**Comment by row49382** [comment]
@jason-edstrom 

`getIndexed()` also wasn't in our parity goal when we initially started developing but I'll go ahead and create that method too.

**Comment by jason-edstrom** [comment]
Perfect.  I was messing around with it this weekend and I have an initial branch started.  It is branching off of this branch.  Feel free to use anything from my branch.

[Lazy Element Enhancements Branch](https://github.com/Magenic/JMAQS/tree/feature/Lazy-Element-Enhancements)

File: jmaqs-selenium/src/main/java/com/magenic/jmaqs/selenium/UIWait.java [hunk 2]
```diff
@@ -1,15 +1,17 @@
 /*
- * Copyright 2020 (C) Magenic, All rights Reserved
```
Comment (jonreding2010) [inline jmaqs-selenium/src/main/java/com/magenic/jmaqs/selenium/UIWait.java:2]: copyright back to 2020

**Comment by row49382** [comment]
@jason-edstrom @billros 

I implemented the requested features. Please review when you find time.

**Pull request merged**